packages/discovery-provider/integration_tests/tasks/test_index_user_bank.py

```diff
@@ -0,0 +1,657 @@
+        # For transfers, the metadata is the source address
+        assert owner_transaction_record.tx_metadata == trackBuyerUserBank
+
+
+# Simulates buying a track for the correct price ($2) but giving one of the recipients
+# an insufficient amount. track owner receives $1.50 and second split receives $0.50 when
+# they should both receive $1
+def test_process_user_bank_tx_details_invalid_purchase_bad_splits(app):
+    tx_response = mock_invalid_track_purchase_bad_splits_tx
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    solana_client_manager_mock = create_autospec(SolanaClientManager)
+
+    transaction = tx_response.value.transaction.transaction
+
+    tx_sig_str = str(transaction.signatures[0])
+
+    challenge_event_bus = create_autospec(ChallengeEventBus)
+
+    populate_mock_db(db, test_entries)
+
+    with db.scoped_session() as session:
+        process_user_bank_tx_details(
+            solana_client_manager=solana_client_manager_mock,
+            redis=redis,
+            session=session,
+            tx_info=tx_response,
+            tx_sig=tx_sig_str,
+            timestamp=datetime.now(),
+            challenge_event_bus=challenge_event_bus,
+        )
+
+        # Expect no purchase record
+        purchase = (
+            session.query(USDCPurchase)
+            .filter(USDCPurchase.signature == tx_sig_str)
+            .first()
+        )
+        assert purchase is None
+
+        # We do still expect the transfers to get indexed, but as regular transfers
+        owner_transaction_record = (
+            session.query(USDCTransactionsHistory)
+            .filter(USDCTransactionsHistory.signature == tx_sig_str)
+            .filter(USDCTransactionsHistory.user_bank == trackOwnerUserBank)
+            .first()
+        )
+        assert owner_transaction_record is not None
+        # Regular transfer, not a purchase
+        assert owner_transaction_record.transaction_type == USDCTransactionType.transfer
+        assert owner_transaction_record.method == USDCTransactionMethod.receive
+        assert owner_transaction_record.change == 1500000
+        # For transfers, the metadata is the source address
+        assert owner_transaction_record.tx_metadata == trackBuyerUserBank
+
+        buyer_transaction_record = (
+            session.query(USDCTransactionsHistory)
+            .filter(USDCTransactionsHistory.signature == tx_sig_str)
+            .filter(USDCTransactionsHistory.user_bank == trackBuyerUserBank)
+            .first()
+        )
+        assert buyer_transaction_record is not None
+        # Regular transfer, not a purchase
+        # This is a little odd in that we index one transfer of $2 out of the sender
+        # account with just the track owner user bank as a destination. Technically
+        # since it's just metadata, this should be fine. But we eventually should include
+        # all addresses, or register a separate transaction for each transfer.
```

Comment:
good callout. I like the multi-transaction solution as it's the easiest on our UI